### PR TITLE
Support Optional Spark features

### PR DESCRIPTION
**Pull request by akhurana001**
# TL;DR
Add support for platform supported optional spark features.  Platform Spark config now supports features and conf to enable for those features. Users can enable or disable features by setting a single spark config and the platform handles the actual configs/implementation etc.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/500

## Follow-up issue
_NA_

### CODE REVIEW

**Review by kumare3 — APPROVED** [review]
lgtm

File: go/tasks/plugins/k8s/spark/spark.go [hunk 1]
```diff
@@ -184,6 +199,27 @@ func (sparkResourceHandler) BuildResource(ctx context.Context, taskCtx pluginsCo
...
+	matches := featureRegex.FindAllStringSubmatch(key, -1)
+	if len(matches) == 0 {
+		return
+	}
+	featureName := matches[0][len(matches[0])-1]
+
+	for _, feature := range GetSparkConfig().Features {
+		if feature.Name == featureName {
+			for k, v := range feature.SparkConfig {
+				sparkConfig[k] = v
```
Comment (EngHabu) [inline go/tasks/plugins/k8s/spark/spark.go:217]: So if multiple features override the same spark configs, the last one wins... is that ok/intentional/undesired (so we can add validation)? whichever it's, I would add docs on the config to explain the behavior

File: go/tasks/plugins/k8s/spark/spark.go [hunk 2]
```diff
@@ -184,6 +199,27 @@ func (sparkResourceHandler) BuildResource(ctx context.Context, taskCtx pluginsCo
 	return j, nil
 }
 
+func addConfig(sparkConfig map[string]string, key string, value string) {
+
+	if strings.TrimSpace(value) != "true" {
```
[the remark this answers is not in the document]
Comment (akhurana001) [reply go/tasks/plugins/k8s/spark/spark.go]: done

File: go/tasks/plugins/k8s/spark/spark.go [hunk 3]
```diff
@@ -184,6 +199,27 @@ func (sparkResourceHandler) BuildResource(ctx context.Context, taskCtx pluginsCo
...
+		return
+	}
+
+	matches := featureRegex.FindAllStringSubmatch(key, -1)
+	if len(matches) == 0 {
```
[the remark this answers is not in the document]
Comment (akhurana001) [reply go/tasks/plugins/k8s/spark/spark.go]: done

File: go/tasks/plugins/k8s/spark/spark.go (hunk 1 above)
Comment (akhurana001) [reply go/tasks/plugins/k8s/spark/spark.go:218]: Ideally they should be unique so this can be a nested map here instead of list of features.  In future thou, I can see this being extended to check for values as well i.e. feature name + value can be unique so hence a struct makes better sense. 

For now, I think using the first matching config makes sense (instead of the last). Added that with comments.

**Review by EngHabu — APPROVED** [review]
